src/inline-assembly.md

```diff
@@ -37,515 +50,892 @@ assert_eq!(x, 4 * 6);
 # }
 ```
 
-## Syntax
+## Syntax [asm.syntax]
```

Comment:
It's a little unclear to me what these bracketed things are in the headers. Can you say more about why you added those, or what their intent is?

Like here, I would expect this to be something like:

```
## Syntax

r[asm.syntax]
```

Reply:
That seems to me like it would be formatted weird. The Rule would appear dangling, not attached to anything.

The rules in the header are also supposed to be the anchor for the entire header section (although I think I got the cmark syntax wrong for that).

Reply:
What does the `r` in front of the bracket refer to? Rule? Render? Where in the md-book documentation can I find that - I tried to search, but couldn't find it?

I am pretty indifferent on which way we go here. The anchor will still basically go to the top even if we format it as @ehuss suggests. I could almost argue another way that it should *above* the header similar to how we do other things.

```
r[asm.syntax]
## Syntax
```

But I don't feel strongly. Let's just pick something and go with it for now.

Reply:
Rule I believe - it's in mdbook-spec not mdbook proper.

Putting the anchor above doesn't quite do what I want, though - I want to replace the default anchor for the header, and I also don't really want the rendering of the rule in this case - it seems like it's supposed to have normal text below it (not a header or a code block), but we either hard switch the context with a header or we put another rule in immediately for the  next actual text entry. That's what I meant by "The Rule would appear dangling".

Reply:
> Where in the md-book documentation can I find that - I tried to search, but couldn't find it?

BTW, it is documented here: https://github.com/rust-lang/reference/blob/master/docs/authoring.md#rules

Reply:
I've changed this to have the rule after. I'd like to have a more extended discussion about this, but it's a mechanical change to revert it anyways so I'm not going to block this PR on it.  I'll open a policy issue for it though, and would like to put it on the agenda for tomorrow. Should that issue go on rust-lang/reference or rust-lang/spec?

Reply:
File [spec#71](https://github.com/rust-lang/spec/issues/71)